src/modules/replay/Replay.cpp

```diff
@@ -149,37 +182,57 @@ Replay::setUserParams(const char *filename)
 		mystrstream >> pname;
 		mystrstream >> value_string;
 
-		double param_value_double = stod(value_string);
-
-		param_t handle = param_find(pname.c_str());
-		param_type_t param_format = param_type(handle);
 		_overridden_params.insert(pname);
 
-		if (param_format == PARAM_TYPE_INT32) {
-			int32_t orig_value = 0;
-			param_get(handle, &orig_value);
+		double param_value_double = stod(value_string);
 
-			int32_t value = (int32_t)param_value_double;
+		setParameter(pname, param_value_double);
+	}
+}
 
-			if (orig_value != value) {
-				PX4_WARN("setting %s (INT32) %d -> %d", param_name(handle), orig_value, value);
-			}
+void
+Replay::readDynamicParams(const char *filename)
+{
+	_dynamic_parameter_schedule.clear();
 
-			param_set(handle, (const void *)&value);
+	string line;
+	string param_name;
+	string value_string;
+	string time_string;
+	ifstream myfile(filename);
 
-		} else if (param_format == PARAM_TYPE_FLOAT) {
-			float orig_value = 0;
-			param_get(handle, &orig_value);
+	if (!myfile.is_open()) {
+		return;
+	}
 
-			float value = (float)param_value_double;
+	PX4_INFO("Reading dynamic params from %s...", filename);
 
-			if (fabsf(orig_value - value) > FLT_EPSILON) {
-				PX4_WARN("setting %s (FLOAT) %.3f -> %.3f", param_name(handle), (double)orig_value, (double)value);
-			}
+	while (!myfile.eof()) {
+		getline(myfile, line);
 
-			param_set(handle, (const void *)&value);
+		if (line.empty() || line[0] == '#') {
+			continue;
 		}
+
+		istringstream mystrstream(line);
+		mystrstream >> param_name;
+		mystrstream >> value_string;
+		mystrstream >> time_string;
+
+		_dynamic_parameters.insert(param_name);
+
+		double param_value = stod(value_string);
+		uint64_t change_timestamp = (uint64_t)(stod(time_string) * 1e6);
+
+		// Construct and store parameter change event
+		tuple<uint64_t, string, double> change_event = make_tuple(change_timestamp, param_name, param_value);
```

Comment:
Using a struct instead of a vector might leave less room for possible errors (no need to remember the order)

Reply:
Good point! I've made the switch.